src/main/java/uk/gov/hmcts/reform/cwrdapi/service/ExcelAdaptorServiceImpl.java

```diff
@@ -0,0 +1,177 @@
+package uk.gov.hmcts.reform.cwrdapi.service;
+
+import org.apache.commons.lang3.tuple.Triple;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.cwrdapi.advice.ExcelValidationException;
+import uk.gov.hmcts.reform.cwrdapi.util.MappingField;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import static java.lang.Boolean.TRUE;
+import static java.util.Objects.nonNull;
+import static org.apache.commons.lang3.BooleanUtils.negate;
+import static org.springframework.core.annotation.AnnotationUtils.findAnnotation;
+import static org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR;
+
+@Service
+@SuppressWarnings("unchecked")
+public class ExcelAdaptorServiceImpl implements ExcelAdaptorService {
+    public static String FILE_NO_DATA_ERROR_MESSAGE = "No data in Excel File";
+    public static String ERROR_FILE_PARSING_ERROR_MESSAGE = "Error while parsing ";
+    public static String IS_PRIMARY_FIELD = "isPrimary";
+    public static String DELIMITER_COMMA = ",";
+
+    public List<Object> parseExcel(Workbook workbook, Class classType) {
+        Sheet sheet = workbook.getSheetAt(1);
+        if (sheet.getPhysicalNumberOfRows() < 2) { // check at least 1 row
```

Comment:
if excel has no sheets may be possible for NullPointerException here ?

Reply:
Done